xir/program.py

```diff
@@ -137,27 +137,44 @@ def use_floats(self) -> bool:
         return self._use_floats
 
 
+class ObservableFactor:
+    def __init__(self, name: str, params: [], wires: [int]):
+        self.name = name
+        self.params = params
+        self.wires = wires
+
+    def __str__(self):
```

Comment:
This method should have a docstring.
```suggestion
    def __str__(self) -> str:
```